app/upgrades/v17/upgrades.go

```diff
@@ -133,3 +150,46 @@ func CreateUpgradeHandler(
 		return migrations, nil
 	}
 }
+
+func FlipTwapSpotPriceRecords(ctx sdk.Context, poolIds []uint64, keepers *keepers.AppKeepers) error {
+	twapRecordHistoricalTimeIndexed, err := keepers.TwapKeeper.GetAllHistoricalTimeIndexedTWAPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, historicalTwapRecord := range twapRecordHistoricalTimeIndexed {
+		oldRecord := historicalTwapRecord
+		historicalTwapRecord.Asset0Denom, historicalTwapRecord.Asset1Denom = oldRecord.Asset1Denom, oldRecord.Asset0Denom
+		historicalTwapRecord.P0LastSpotPrice, historicalTwapRecord.P1LastSpotPrice = oldRecord.P1LastSpotPrice, oldRecord.P0LastSpotPrice
+
+		keepers.TwapKeeper.StoreHistoricalTWAP(ctx, historicalTwapRecord)
+		keepers.TwapKeeper.DeleteHistoricalRecord(ctx, oldRecord)
+	}
```

Comment:
As it stands right now, historical records for all pools end up being switched. We only want to do so for the OSMO / DAI  CL pool.

What we should to instead is:
1. iterate the historical pool indexed twaps, providing the pool id of the desired pool
2. for every record do (as it is done now with the exception of only the desired pool twaps are changed):
  a. switch fields
  b. store updated historical twap
  c. delete old

Reply:
Additionally, would like tests to be added, making sure that records for other pools are not updated